js/utils.js

```diff
@@ -1300,12 +1301,20 @@ Fliplet.Registry.set('dynamicListUtils', (function() {
           field: field,
           useData: true
         }), function(item) {
+          var dates = records.map(function(obj) {
```

Comment:
This is already happening inside a `records.forEach()` loop, which means we're running a loop inside a loop. If the data source has 100,000 entries, which regularly happens, this would result in a lot of unnecessary processing time.